# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/fetch/FetchParent.h
# Commit: b9d73f961292
# Full Hash: b9d73f9612929dda88ca7e68d8d2e04de2aa2a86
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2023-01-20 21:21:03
# Regressor Bug: 1351231
# File Overlap Count: 1
# Description:
#   Bug 1351231 - FetchService integration for PFetch. r=dom-worker-reviewers,jesup
#   
#   Depends on D142436
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D142437
# ==============================================================================

diff -r 989cbb08b87e -r b9d73f961292 dom/fetch/FetchParent.h
--- a/dom/fetch/FetchParent.h	Fri Jan 20 09:09:17 2023 +0000
+++ b/dom/fetch/FetchParent.h	Fri Jan 20 09:09:17 2023 +0000
@@ -7,19 +7,23 @@
 
 #include "mozilla/Maybe.h"
 #include "mozilla/MozPromise.h"
+#include "mozilla/Mutex.h"
 #include "mozilla/RefPtr.h"
-#include "mozilla/dom/FetchService.h"
 #include "mozilla/dom/PFetchParent.h"
 #include "mozilla/dom/SafeRefPtr.h"
 #include "mozilla/ipc/PBackgroundSharedTypes.h"
+#include "mozilla/net/NeckoChannelParams.h"
 #include "nsCOMPtr.h"
 #include "nsIContentSecurityPolicy.h"
+#include "nsID.h"
 #include "nsISerialEventTarget.h"
 #include "nsString.h"
+#include "nsTHashMap.h"
 
 namespace mozilla::dom {
 
 class ClientInfo;
+class FetchServicePromises;
 class InternalRequest;
 class InternalResponse;
 class ServiceWorkerDescriptor;
@@ -38,7 +42,14 @@
 
   static RefPtr<FetchParent> GetActorByID(const nsID& aID);
 
-  void OnFlushConsoleReport(nsTArray<net::ConsoleReportCollected>&& aReports);
+  void OnResponseAvailableInternal(SafeRefPtr<InternalResponse>&& aResponse);
+
+  void OnResponseEnd(const ResponseEndArgs& aArgs);
+
+  void OnDataAvailable();
+
+  void OnFlushConsoleReport(
+      const nsTArray<net::ConsoleReportCollected>& aReports);
 
   class FetchParentCSPEventListener final : public nsICSPEventListener {
    public:
@@ -76,9 +87,11 @@
   nsCString mWorkerScript;
   Maybe<ClientInfo> mClientInfo;
   Maybe<ServiceWorkerDescriptor> mController;
-  RefPtr<FetchParentCSPEventListener> mCSPEventListener;
+  Maybe<CookieJarSettingsArgs> mCookieJarSettings;
+  nsCOMPtr<nsICSPEventListener> mCSPEventListener;
   bool mNeedOnDataAvailable{false};
   bool mHasCSPEventListener{false};
+  bool mExtendForCSPEventListener{false};
 
   Atomic<bool> mIsDone{false};
   Atomic<bool> mActorDestroyed{false};